### PR TITLE
A0-2082: Merge `snarkeling` to `main`

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# DO NOT SQUASH COMMITS (requires allowance)

# Description

Putting all the liminal code behind corresponding features. Liminal workflows will be done in subsequent PRs.

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have created new documentation
- I have bumped `spec_version` and `transaction_version`
- I have bumped aleph-client version if relevant

### CODE REVIEW

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -325,6 +327,21 @@ impl pallet_aleph::Config for Runtime {
     type NextSessionAuthorityProvider = Session;
 }
 
+#[cfg(feature = "liminal")]
+parameter_types! {
```
Comment (Marcin-Radecki) [inline bin/runtime/src/lib.rs:331]: As a matter of sanity check, could you please compare checksum of `aleph-release-runtime` build by this PR and `main` commit ? We expect those to be the same

**Review by Marcin-Radecki — APPROVED** [review]
Splendid work!

File: bin/runtime/Cargo.toml [hunk 2]
```diff
@@ -55,12 +55,19 @@ sp-std = { default-features = false, git = "https://github.com/Cardinal-Cryptogr
 sp-transaction-pool = { default-features = false, git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.38" }
 sp-version = { default-features = false, git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.38" }
 # NOTE: these two disabled features collide with std in wasm
-sp-io = { default-features = false, git = "https://github.com/Cardinal-Cryptography/substrate.git", branch = "aleph-v0.9.38" }
+sp-io = { default-features = false, features = ["disable_oom", "disable_panic_handler"], git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.38" }
```
[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply bin/runtime/Cargo.toml]: good catch - I reverted this line from long ago, when we didn't maintain fork - fixed

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -4,8 +4,15 @@ name: unit-tests
 on:
   pull_request:
     paths-ignore:
+      - 'aleph-client/**'
```
[the remark this answers is not in the document]
Comment (pmikolajczyk41) [reply .github/workflows/unit_tests.yml:7]: thanks, fixed

**Review by mike1729 — APPROVED** [review]

File: bin/runtime/src/lib.rs (hunk 1 above)
Comment (Marcin-Radecki) [reply bin/runtime/src/lib.rs:331]: Actually due to common workspace Cargo.lock there are changes expected.

**Pull request merged**